src/app/candidate-management/candidate-management-detail/candidate-management-detail.component.ts

```diff
@@ -60,47 +62,66 @@ export class CandidateManagementDetailComponent implements OnInit, AfterViewInit
   ) { }
 
   ngOnInit(): void {
-
-    this.candidateStore.candidate.subscribe((_candidate: Candidate) => {
-      if (_candidate && this.router.url.includes('detail')) {
-        this.disabled = true;
-        this.candidate = _candidate;
-        this.contentToMarkdown();
-        this.checkTreshhold();
-
-      } else if (_candidate && this.router.url.includes('edit')) {
-        this.candidate = _candidate;
-        this.contentToMarkdown();
-        this.edit();
-        this.checkTreshhold();
-
-      } else if (!_candidate && window.history.state.data && window.history.state.data instanceof Candidate) {
-        this.candidate = window.history.state.data as Candidate
-        this.contentToMarkdown();
-        this.edit();
-        this.checkTreshhold();
-
-      } else {
-        this.disabled = false;
-        this.candidate = new Candidate(null, 'New Candidate', null, null);
-        this.patternLanguageSelectedChange(patternLanguageNone);
-        // Preset author
-        this.auth.user.subscribe(_user => {
-          if (_user && !this.candidate.authors) this.candidate.authors = [new AuthorModel(_user.id, Author.OWNER, _user.name)];
-        })
-      }
-      this.confirmDialog = {
-        title: `Change Pattern Language for Candidate ${this.candidate.name}`,
-        text: 'If you change the language everything writen will be deleted and the'
-          + ' new pattern schema will be used'
+    this.arSubscription = this.activeRoute.params.subscribe(params => {
+      let candidateUri = `/candidates/${params.name}`;
+      switch (params.action) {
+        case 'detail': {
+          this.disabled = true;
+          this.candidateManagementService.getCandidateByUri(candidateUri).subscribe(result => {
+            this.candidate = result;
+            this.contentToMarkdown();
+            this.checkTreshhold();
+            this.setupPLChangeDialog();
+          });
+          break;
+        }
+        case 'edit': {
+          this.candidateManagementService.getCandidateByUri(candidateUri).subscribe(result => {
```

Comment:
If I switch from detail to edit mode I may not need to reload at all if the candidate name did not change as well. (of course this would complicate the logic)